protocol/messenger.go

```diff
@@ -3009,6 +3012,101 @@ func (m *Messenger) outputToCSV(timestamp uint32, messageID types.HexBytes, from
 	}
 }
 
+func (m *Messenger) handleImportedMessages(messagesToHandle map[transport.Filter][]*types.Message) error {
```

Comment:
@cammellos this is related to our discussion offline re: using a third party ID to handle imported messages differently (and not passing them to `handleRetrievedMessages()`.

Notice that the function signature of `handleImportedMessages()` is the same.
Also, the functionality is really just a trimmed down version of what `handleRetrievedMessages()` does, plus, it sets the `messageID = `msg.TransportMessage.ThirdPartyID`

We could consider using `handleRetrievedMessage()` instead and simply override `messageID` with `ThirdPartyID` if it exists. Not entirely sure if adding `handleImportMessages()` adds way more value then otherwise.